### DISCUSSION TITLE
Can not connect

### DISCUSSION THREAD

**Discussion by ruchan-ziya** [vinodsr/node-red-contrib-tuya-smart-device#119] [Q&A]
Hi,

I saw that there are other posts with the same question that has not been answered for more than a year, but I wanted to try my chances.. I obtained the DeviceID and Localkey values. And tested that they work in the API Explorer. 

I entered these values in the node-red node but all I am getting is "find() timed out. Is the device powered on and the ID or IP correct?"

I noticed that the API explorer in the Tuya Portal has authentication parameters in the call, It says "..the platform uses the authorization key (Access ID & Access Secret) of the current project to get a temporary token and initiate API calls to access your data and online resources..." 

But I could not find a way to enter any authentication info for node-red tuya-smart-device . Is the pair the device-id and local-key sufficient to connect? Or do we need to authenticate first somehow? If they are sufficient what else I could be doing wrong?

Thank you for your help.
Ruchan

**Comment by tenzeroauto** [comment]
Hi Ruchan,
I was getting the same issue.
My solution was to add the IP address as well as the DeviceID and Device Key
 "deviceVirtualId": "IDIDIDIDIDIDIDIDIDIDIDIDIDID",
    "deviceKey": "KKKKKKKKKKKKKKKK",
    "deviceIp": "xxx.xxx.xxx.xxx",

**Reply by ruchan-ziya** [reply]
Does that mean the device and the node-red has to be on the same network? I was hoping node-red would connect over the internet. 

There is an IP on the tuya portal for the device but it is the Internet IP address the router has. I tried it anyway but failed.  This time
`"Error: Error from socket: connect ECONNREFUSED XXX.XXX.XXX.XXX:6668"`

Could port mapping be a solution? I could map 6668 in the router to the device...  But then sometimes router reboots and gets a new IP. This would break things,...
